clients/client/src/main/java/org/projectnessie/client/auth/AwsAuthenticationProvider.java

```diff
@@ -78,15 +84,18 @@ private static Region awsRegion(String regionName) {
 
   private static class AwsAuthentication implements HttpAuthentication {
     private final Region region;
+    private final AwsCredentialsProvider awsCredentialsProvider;
 
     @SuppressWarnings({"QsPrivateBeanMembersInspection", "CdiInjectionPointsInspection"})
-    private AwsAuthentication(Region region) {
+    private AwsAuthentication(Region region, String profile) {
       this.region = region;
+      this.awsCredentialsProvider =
+          DefaultCredentialsProvider.builder().profileName(profile).build();
```

Comment:
Not sure whether it's worth to special-case `profile==null` to prevent that it may throw an NPE when they change the builder impl? WDYT?